diff --git a/src/gpu/ganesh/ops/AtlasPathRenderer.cpp b/src/gpu/ganesh/ops/AtlasPathRenderer.cpp
index 2a31b5a0d8..943bd29ece 100644
--- a/src/gpu/ganesh/ops/AtlasPathRenderer.cpp
+++ b/src/gpu/ganesh/ops/AtlasPathRenderer.cpp
@@ -106,6 +106,13 @@ bool AtlasPathRenderer::IsSupported(GrRecordingContext* rContext) {
     if (rContext->backend() == GrBackendApi::kOpenGL) {
         return false;
     }
+#endif
+#ifdef SK_BUILD_FOR_WIN
+    // http://skbug.com/13519 There is a bug with the atlas path renderer on Direct3D, running on
+    // Radeon hardware and possibly others. Disable until we can investigate.
+    if (rContext->backend() == GrBackendApi::kDirect3D) {
+        return false;
+    }
 #endif
     const GrCaps& caps = *rContext->priv().caps();
     auto atlasFormat = caps.getDefaultBackendFormat(kAtlasAlpha8Type, GrRenderable::kYes);
