diff --git a/modules/skparagraph/src/OneLineShaper.cpp b/modules/skparagraph/src/OneLineShaper.cpp
index b6a91c5..0b13f93 100644
--- a/modules/skparagraph/src/OneLineShaper.cpp
+++ b/modules/skparagraph/src/OneLineShaper.cpp
@@ -487,6 +487,7 @@
     if (fParagraph->fFontCollection->fontFallbackEnabled()) {
         // Give fallback a clue
         // Some unresolved subblocks might be resolved with different fallback fonts
+        std::vector<RunBlock> hopelessBlocks;
         while (!fUnresolvedBlocks.empty()) {
             auto unresolvedRange = fUnresolvedBlocks.front().fText;
             auto unresolvedText = fParagraph->text(unresolvedRange);
@@ -527,6 +528,8 @@
 
                 if (ch == unresolvedText.end()) {
                     // Not a single codepoint could be resolved but we finished the block
+                    hopelessBlocks.push_back(fUnresolvedBlocks.front());
+                    fUnresolvedBlocks.pop_front();
                     break;
                 }
 
@@ -540,7 +543,10 @@
                     }
                 }
             }
+        }
 
+        for (auto& block : hopelessBlocks) {
+            fUnresolvedBlocks.emplace_front(block);
         }
     }
 }
